pkg/plugins/driver.proto

```diff
@@ -0,0 +1,176 @@
+syntax = "proto3";
+
+package plugins;
+
+option go_package = "github.com/lima-vm/lima/pkg/plugins";
+
+service VMDriver {
+    rpc GetMetadata(GetMetadataRequest) returns (GetMetadataResponse);
+    rpc StartVM(StartVMRequest) returns (StartVMResponse);
+    rpc StopVM(StopVMRequest) returns (StopVMResponse);
+    rpc Initialize(InitializeRequest) returns (InitializeResponse);
+    rpc CreateDisk(CreateDiskRequest) returns (CreateDiskResponse);
+    rpc Validate(ValidateRequest) returns (ValidateResponse);
+    rpc Register(RegisterRequest) returns (RegisterResponse);
+    rpc Unregister(UnregisterRequest) returns (UnregisterResponse);
+    rpc ChangeDisplayPassword(ChangeDisplayPasswordRequest) returns (ChangeDisplayPasswordResponse);
+    rpc GetDisplayConnection(GetDisplayConnectionRequest) returns (GetDisplayConnectionResponse);
+    rpc CreateSnapshot(CreateSnapshotRequest) returns (CreateSnapshotResponse);
+    rpc ApplySnapshot(ApplySnapshotRequest) returns (ApplySnapshotResponse);
+    rpc DeleteSnapshot(DeleteSnapshotRequest) returns (DeleteSnapshotResponse);
+    rpc ListSnapshots(ListSnapshotsRequest) returns (ListSnapshotsResponse);
+    rpc GetGuestAgentConnection(GetGuestAgentConnectionRequest) returns (GetGuestAgentConnectionResponse);
+}
+
+// Response is a common response type used by helper functions
+message Response {
+    bool success = 1;
```

Comment:
Probably `Response` should not be returned on an error?

gRPC has its own error reporting system
https://github.com/grpc/grpc-go/blob/6819ed796fcd0232a46dab21c1b7826aa7f1d561/examples/features/error_details/client/main.go#L55-L65